crates/db_connection_pool/src/dbconnection/odbcconn.rs

```diff
@@ -173,6 +174,26 @@ where
         let params = params.iter().map(dyn_clone::clone).collect::<Vec<_>>();
         let secrets = Arc::clone(&self.params);
 
+        let cancellation_token = tokio_util::sync::CancellationToken::new();
+        let cloned_token = cancellation_token.clone();
+
+        tokio::spawn(async move {
+            let ctrl_c = async {
+                let signal_result = signal::ctrl_c().await;
+                if let Err(err) = signal_result {
+                    tracing::error!(
+                        "Failed to listen to shutdown signal in ODBC data fetching: {err}"
+                    );
+                } else {
+                    cancellation_token.clone().cancel();
```

Comment:
Move this cancel outside of the else - even if listening fails, we still want to cancel